optimum/bettertransformer/models/encoder_models.py

```diff
@@ -343,6 +343,114 @@ def forward(self, hidden_states, attention_mask, position_bias=None, *_, **__):
             hidden_states = hidden_states.to_padded_tensor(0.0)
         return (hidden_states,)
 
+class MBartEncoderLayerBetterTransformer(BetterTransformerBaseLayer):
+    def __init__(self, mbart_layer, config):
+        r"""
+        A simple conversion of the `MBartEncoderLayer` to its `BetterTransformer` implementation.
+        Args:
+            mbart_layer (`torch.nn.Module`):
+                The original `MBartEncoderLayer` where the weights needs to be retrieved.
+        """
+        super().__init__(config)
+        # In_proj layer
+        self.in_proj_weight = nn.Parameter(
+            torch.cat(
+                [
+                    mbart_layer.self_attn.q_proj.weight,
+                    mbart_layer.self_attn.k_proj.weight,
+                    mbart_layer.self_attn.v_proj.weight,
+                ]
+            )
+        )
+
+        self.in_proj_bias = nn.Parameter(
+            torch.cat(
+                [
+                    mbart_layer.self_attn.q_proj.bias,
+                    mbart_layer.self_attn.k_proj.bias,
+                    mbart_layer.self_attn.v_proj.bias,
+                ]
+            )
+        )
+
+        # Out proj layer
+        self.out_proj_weight = mbart_layer.self_attn.out_proj.weight
+        self.out_proj_bias = mbart_layer.self_attn.out_proj.bias
+
+        # Linear layer 1
+        self.linear1_weight = mbart_layer.fc1.weight
+        self.linear1_bias = mbart_layer.fc1.bias
+
+        # Linear layer 2
+        self.linear2_weight = mbart_layer.fc2.weight
+        self.linear2_bias = mbart_layer.fc2.bias
+
+        # Layer norm 1
+        self.norm1_eps = mbart_layer.self_attn_layer_norm.eps
+        self.norm1_weight = mbart_layer.self_attn_layer_norm.weight
+        self.norm1_bias = mbart_layer.self_attn_layer_norm.bias
+
+        # Layer norm 2
+        self.norm2_eps = mbart_layer.final_layer_norm.eps
+        self.norm2_weight = mbart_layer.final_layer_norm.weight
+        self.norm2_bias = mbart_layer.final_layer_norm.bias
+
+        # Model hyper parameters
+        self.num_heads = mbart_layer.self_attn.num_heads
+        self.embed_dim = mbart_layer.self_attn.embed_dim
+
+        # Last step: set the last layer to `False` -> this will be set to `True` when converting the model
+        self.is_last_layer = False
+
```

Comment:
```suggestion
        self.norm_first = True
```
As you can see [here](https://github.com/huggingface/transformers/blob/95754b47a6d4fbdad3440a45762531e8c471c528/src/transformers/models/mbart/modeling_mbart.py#L332), `MBart` seems to use pre-attention layer norm. Setting this attribute to `True` should fix the issue